packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts

```diff
@@ -0,0 +1,252 @@
+    });
+
+    CfnParameters.unauthRoleName = new cdk.CfnParameter(this, 'UnauthRoleName', {
+      type: 'String',
+      default: 'UnauthRoleName',
+    });
+    // Reources
+
+    let CfnResources: AmplifyRootStackResourceProps = {};
+    CfnResources.DeploymentBucket = new s3.CfnBucket(this, 'DeploymentBucket', {
+      bucketName: CfnParameters['deploymentBucketName'].valueAsString,
+    });
+
+    CfnResources.DeploymentBucket.applyRemovalPolicy(cdk.RemovalPolicy.RETAIN);
+
+    CfnResources.authRole = new iam.CfnRole(this, 'AuthRole', {
+      roleName: CfnParameters['authRoleName'].valueAsString,
+      assumeRolePolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Sid: '',
+            Effect: 'Deny',
+            Principal: {
+              Federated: 'cognito-identity.amazonaws.com',
+            },
+            Action: 'sts:AssumeRoleWithWebIdentity',
+          },
+        ],
+      },
+    });
+
+    CfnResources.unauthRole = new iam.CfnRole(this, 'UnauthRole', {
+      roleName: CfnParameters['unauthRoleName'].valueAsString,
+      assumeRolePolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Sid: '',
+            Effect: 'Deny',
+            Principal: {
+              Federated: 'cognito-identity.amazonaws.com',
+            },
+            Action: 'sts:AssumeRoleWithWebIdentity',
+          },
+        ],
+      },
+    });
+
+    const cfnOutputs: { [k: string]: cdk.CfnOutput } = {};
+    cfnOutputs.Region = new cdk.CfnOutput(this, 'Region', {
+      description: 'CloudFormation provider root stack Region',
+      value: cdk.Fn.ref('AWS::Region'),
+      exportName: cdk.Fn.sub('${AWS::StackName}-Region'),
+    });
+
+    cfnOutputs.stackName = new cdk.CfnOutput(this, 'StackName', {
+      description: 'CloudFormation provider root stack ID',
+      value: cdk.Fn.ref('AWS::StackName'),
+      exportName: cdk.Fn.sub('${AWS::StackName}-StackName'),
+    });
+
+    cfnOutputs.stackId = new cdk.CfnOutput(this, 'StackId', {
+      description: 'CloudFormation provider root stack name',
+      value: cdk.Fn.ref('AWS::StackId'),
+      exportName: cdk.Fn.sub('${AWS::StackName}-StackId'),
+    });
+
+    cfnOutputs.authRoleArn = new cdk.CfnOutput(this, 'AuthRoleArn', {
+      value: cdk.Fn.getAtt('AuthRole', 'Arn').toString(),
+    });
+
+    cfnOutputs.unauthRoleArn = new cdk.CfnOutput(this, 'UnauthRoleArn', {
+      value: cdk.Fn.getAtt('UnauthRole', 'Arn').toString(),
+    });
+    this.templateObj = {};
+    this.templateObj['Parameters'] = { ...CfnParameters };
+    this.templateObj['Outputs'] = { ...cfnOutputs };
+    this.templateObj.Resources = { ...CfnResources };
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+    return this._toCloudFormation();
+  }
+}
+
+export enum CommandType {
+  'PUSH',
+  'INIT',
+}
+
+type RootStackOptions = {
+  rootStackFileName: string;
+  event: CommandType;
+  overrideDir?: string;
+};
+
+/**
```

Comment:
Could you elaborate

Reply:
In the Root stack, we have parameters and outputs with same logicalID as ```deployment bucketName```. While constructing cdk constructs , we have have resources in stack with same logical ID. To maintain consistency with existing projects , I created another constructs having duplicate logicalIDs present and merge the both templates after synthesizing it.